runtime/js-compute-runtime/builtins/fastly.cpp

```diff
@@ -130,6 +131,70 @@ bool Fastly::includeBytes(JSContext *cx, unsigned argc, JS::Value *vp) {
   return true;
 }
 
+bool Fastly::createFanoutHandoff(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = CallArgsFromVp(argc, vp);
+  REQUEST_HANDLER_ONLY("createFanoutHandoff");
+  if (!args.requireAtLeast(cx, "createFanoutHandoff", 2)) {
+    return false;
+  }
+
+  auto request_value = args.get(0);
+  if (!Request::is_instance(request_value)) {
+    JS_ReportErrorUTF8(cx, "createFanoutHandoff: request parameter must be an instance of Request");
+    return false;
+  }
+
+  fastly_response_handle_t response_handle = INVALID_HANDLE;
+  fastly_error_t err;
+  if (!fastly_http_resp_new(&response_handle, &err)) {
+    HANDLE_ERROR(cx, err);
+    return false;
+  }
+  fastly_body_handle_t body_handle = INVALID_HANDLE;
+  if (!fastly_http_body_new(&body_handle, &err)) {
+    HANDLE_ERROR(cx, err);
+    return false;
+  }
+
+  JS::RootedObject response_instance(cx, JS_NewObjectWithGivenProto(cx, &builtins::Response::class_,
+                                                                    builtins::Response::proto_obj));
+  if (!response_instance) {
+    return false;
+  }
+
+  auto backend_value = args.get(1);
+  size_t length;
+  auto backend_chars = encode(cx, backend_value, &length);
+  if (!backend_chars) {
+    return false;
+  }
+  if (length == 0) {
+    JS_ReportErrorUTF8(cx, "createFanoutHandoff: Backend parameter can not be an empty string");
+    return false;
+  }
+
+  if (length > 254) {
+    JS_ReportErrorUTF8(cx, "createFanoutHandoff: name can not be more than 254 characters");
+    return false;
+  }
+
+  char *buf = new char[length + 1];
```

Comment:
What do you think about making this a `JS::UniqueChars` instead? That way we'd ensure that ownership is transferred to the request, or that we'd clean it up if we failed to create the response. Additionally, we'd then have an explicit point where we're leaking the memory in `Response::create`, when we would be releasing the memory to write the pointer into the js object that holds it.

Reply:
Additionally, make sure to write the `0` byte at the end if this needs to be null-terminated.

Reply:
`JS::UniqueChars` with `std::move` and `.release()` is a great idea 👍
We can use the `backend_chars` variable above which is a `JS::UniqueChars` of the backend name